### PR TITLE
improve CLI coverage

### PR DESCRIPTION
**Pull request by minrk**
- [x] cover git config entrypoints
- [x] cover more nbmerge options
- [x] exercise merge/diff drivers with a real git repo
- [x] exercise merge/diff tools
- [x] exercise nbmerge/diff-web
- [x] some cleanup in travis / dependencies

### CODE REVIEW

File: nbdime/gitmergetool.py [hunk 1]
```diff
@@ -85,10 +86,12 @@ def main(args=None):
         dest='config_func', const=disable,
         help="disable nbdime mergetool via git config"
     )
-    opts = parser.parse_args(args)
+    opts, extra_args = parser.parse_known_args(args)
     nbdime.log.init_logging(level=opts.log_level)
     if opts.subcommand == 'merge':
-        return nbmergetool.main([opts.base, opts.local, opts.remote, opts.merged])
+        args = list(args)
+        args.remove('merge')
```
Comment (vidartf) [inline nbdime/gitmergetool.py]: 'merge' or 'merged'?

File: nbdime/tests/test_cli_apps.py [hunk 2]
```diff
@@ -34,22 +67,317 @@ def test_nbdiff_app():
...
+
+def test_nbmerge_app_conflict(tempfiles, capsys):
+    p = tempfiles
+    bfn = os.path.join(p, "inline-conflict--1.ipynb")
+    lfn = os.path.join(p, "inline-conflict--2.ipynb")
+    rfn = os.path.join(p, "inline-conflict--3.ipynb")
+    ofn = os.path.join(p, "inline-conflict-out.ipynb")
+
+    assert 1 == nbmergeapp.main([bfn, lfn, rfn])
+    nb_stdout, err = capsys.readouterr()
```
Comment (vidartf) [inline nbdime/tests/test_cli_apps.py]: unused?

**Comment by willingc** [comment]
@minrk http://pastebin.com/LBE0KFd6 re: colorama

**Comment by minrk** [comment]
@willingc looks like nbdime and pytest aren't in the same env (pytest came from `/Library/Frameworks/Python.framework/Versions/3.5`). Try `pip install nbdime[test]` (then `hash -r` since you've already called pytest in the env) first. I just added the `pip install nbdime[test]` command to the README.

**Comment by willingc** [comment]
Thanks @minrk. Working great now.

**Comment by minrk** [comment]
Thanks, @willingc!

**Comment by willingc** [comment]
Yay testing. Code looks good to me. Thanks @minrk.

**Pull request merged**